### ISSUE TITLE
How to run this on cpu

### ISSUE THREAD

**Issue by MrWD10** [LearnedVector/A-Hackers-AI-Voice-Assistant#27]
i want to train it using cpu instead of gpu.under speech recognition folder------>in train.py what should I change in code to work it properly?

**Comment by KnifeOfDunwall** [comment]
It has a line of code that uses gpu if available and cpu if not. I'm currently on my phone and cannot give you specific details where it is but it should be in train.py. You need to change that line so that it always uses cpu

**Comment by MrWD10** [comment]
@KnifeOfDunwall train.py under speech recognition folder does not contain such lines. it will be really great if you could manage some time to look into this issue.

**Comment by MrWD10** [comment]
@KnifeOfDunwall <https://github.com/KnifeOfDunwall> train.py under speech
recognition folder does not contain such lines. it will be really great if
you could manage some time to look into this issue.

**Comment by KnifeOfDunwall** [comment]
youre right, it was in the train.py of the wakeword. try to do [this](https://stackoverflow.com/questions/54544986/need-to-change-gpu-option-to-cpu-in-a-python-pytorch-based-code).

**Comment by MrWD10** [comment]
And what about train.py of speech recognition. They don't use that
method(cuda if device available). I guess by changing value of GPU to 0
from 1 will do the trick...?

**Comment by MrWD10** [comment]
@KnifeOfDunwall will this repo will run on google colab? because some of its libraries do not operate in windows i.e torchaudio.......
when I try to run it in colab I got these errors, don't know why.....can you help me...?
![Screenshot from 2020-11-04 04-56-16](https://user-images.githubusercontent.com/39570348/98053891-780e6a80-1e5b-11eb-9bfa-078ac920e6bd.png)

**Comment by KnifeOfDunwall** [comment]
I dont know how to use colab. you should ask someone else as im new to this

**Comment by MrWD10** [comment]
Then can you tell me how to resolve this issue:
Exception: dual channel, skipping audio file (file_path)
This exception is being generated from the dataset.py file under
speechrecognition folder. The spectrogram is keep returning channel=2 which
causes the above exception to raised. Can you help me solving this issue..?

**Comment by XlentAcademy** [comment]
So my friend, in google colab you have to disable the save_model_path or instead of args.save_model_path you should write '/content/(and here your dir path)'

**Comment by urdaibayc** [comment]
> @KnifeOfDunwall will this repo will run on google colab? because some of its libraries do not operate in windows i.e torchaudio.......
> when I try to run it in colab I got these errors, don't know why.....can you help me...?
> ![Screenshot from 2020-11-04 04-56-16](https://user-images.githubusercontent.com/39570348/98053891-780e6a80-1e5b-11eb-9bfa-078ac920e6bd.png)

[did you do this?]((https://hackernoon.com/getting-started-with-pytorch-in-google-collab-with-free-gpu-61a5c70b86a))

**Comment by XlentAcademy** [comment]
Yes, i have a trained a wakeword model on google colab.
Just use GPU instead of TPU your error will be solved.

**Comment by m0ya** [comment]
could plz tell me how to wakeword model on colab and what are changes?

**Comment by Jochen-sys** [comment]
Hey I have a question left: The changing from gpu to cpu has been for wakeword detection and not for whole sentences?
